Player_Unit.java

```diff
@@ -30,23 +30,16 @@ public Player_Unit(){
 
     public void lvl_up(){
         lvl+=1;
-        xp=xp+xp_per_lvl;
-        hp=hp+hp_per_lvl;
-        atk=atk+dmg_per_lvl;
-        atk_cd=atk_cd*cd_mp_per_lvl;
+        xp+=xp_per_lvl;
+        hp+=hp_per_lvl;
+        atk+=dmg_per_lvl;
+        atk_cd*=cd_mp_per_lvl;
     }
 
     public void attack(Unit defender){
         defender.hp -= getAtk();
-        dmg_inflicted += getAtk();
-        if(dmg_inflicted/dmg_required_to_lvlup>lvl){
-
-            int lvlup_number = (int) (dmg_inflicted/dmg_required_to_lvlup)-lvl;
-
-            for(int i=0; i<lvlup_number;i++){
-                lvl_up();
-            }        
-        }
-    
+        dmg_inflicted = (defender.isAlive()) ? getAtk() : defender.hp+getAtk();
+        int lvlup_number = (int) (dmg_inflicted/dmg_required_to_lvlup)-lvl;
```

Comment:
Ez most nagyon nem tunik jonak. 
1) miert adodik hp-hoz hozza az atk?
2) most nem osszegzodik az XP, csak az egy utesen belulivel lehet szintet lepni.

Ez hogy jott ki? Biztos ezt akartad?

Reply:
Igen az összegződét véletlenül lefelejtettem, de azóta += lett. 

Úgy akartam hogy ugye megvizsgálja a sebzés levonása után hogy él e még a megtámadott unit, ha él akkor a támadó attack pontját adom hozzá a bevitt sebzésekhez. Ha nem él akkor meg a védekező negatív hp-jához hozzá adom a támadó attack pontját és így megkapom a megtámadott unit támadás előtti hp-ját ami elvileg egyenlő a bevitt sebzéssel is. Legalábbis ezt akartam, csak ugye letesztelni nem tudtam, de szerintem működnie kéne.

Reply:
Jah, ezt a logikat igy mar ertem, de akkor igen, +=